### PR TITLE
Gauntlet test - Add consolidate IT when JVM / queue rejection metrics are issued one after another 

### PR DESCRIPTION
**Pull request by rguo-aws**
*Fixes #:*

*Description of changes:*
Add consolidate IT to cover the scenario two deciders are trying to issue two set of actions in different direction. As a result, Flip flop detector will jump in to suppress one set of actions. 

1. we start with low JVM usage and then increase the JVM after 10 seconds. JVM RCA is expected to be triggered and we should be able to observe JVM actions to lower heap usage in about 3-4 mins

2. Once JVM decisions are published, we lower the JVM usage to bring JVM RCA back to normal and immediately inject queue rejection metrics. Flip Flop detector should capture this and suppress queue rejection actions because increasing queue capacity will change JVM vector into a different direction.

*Tests:*
n/a

*If new tests are added, how long do the new ones take to complete*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by yojs — APPROVED** [review]

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java [hunk 1]
```diff
@@ -183,6 +184,17 @@ public void putMetric(Metric<Double> metric, Dimensions dimensions, long windowS
         .execute();
   }
 
+  /**
+   * Drop a metric table. This is for IT framework to use only
+   * @param metricName metric table to be deleted
+   */
+  @VisibleForTesting
```
Comment (yu-sun-77) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java:191]: Do we need `VisibleForTesting` annotation for public method?

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/consolidate_tuning/JvmFlipFlopITest.java [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+                      sum = 500, avg = 500, min = 500, max = 500),
+                  @ATuple(dimensionValues = {ThreadPoolType.Constants.SEARCH_NAME},
+                      sum = 1500, avg = 1500, min = 1500, max = 1500)
+              }
+          )
+      }
+  )
+  public static class MetricsForQueueRejection {
+  }
+}
```
Comment (yu-sun-77) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/consolidate_tuning/JvmFlipFlopITest.java:334]: nit: newline

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/consolidate_tuning/JvmFlipFlopITest.java [hunk 3]
```diff
@@ -0,0 +1,334 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.integTests.tests.consolidate_tuning;
```
Comment (yu-sun-77) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/consolidate_tuning/JvmFlipFlopITest.java:1]: can we add license header?

**Review by yu-sun-77 — APPROVED** [review]
Overall LGTM, some minor comments

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java (hunk 1 above)
Comment (rguo-aws) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java:191]: this function needs to be called outside of MetricDB but is intended to be used by UT only. The metricDB is written only once in prod.

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/consolidate_tuning/JvmFlipFlopITest.java (hunk 3 above)
Comment (rguo-aws) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/consolidate_tuning/JvmFlipFlopITest.java:1]: will add the header in my next PR. Thanks for pointing this out

**Pull request merged**